src/glfw3.jl

```diff
@@ -509,8 +510,11 @@ GLFW.SetWindowIcon(win, buff)
 GLFW.PollEvents() # seems to need a poll events to become active
 ```
 """
-SetWindowIcon(window::Window, image::Matrix{NTuple{4, UInt8}}) = ccall((:glfwSetWindowIcon, lib),
-	Cvoid, (Window, Cint, GLFWImage), window, 1, image)
```

Comment:
I'm really surprised this code even worked ...    the c signature requires a pointer to GLFWImage, however the julia code here passes directly a GLFWImage struct to  the c function.

Reply:
cconvert should do the magic ;)

Reply:
right, but the manual overrides ensured that  a   GLFW struct is directly passes to the signature

```
Base.unsafe_convert(GLFWImage, Base.cconvert(GLFWImage, image))
```
return a GLFWImage which is then passed directly to glfwSetWindowIcon, but this function takes as inputs  pointer to an array with the GLFWImage structure.....

Reply:
Hm I see... maybe should have been wrapped in a Ref, and we got lucky that structs get passed by reference for ccall in any case ;)

Reply:
`Hm I see... maybe should have been wrapped in a Ref, and we got lucky that structs get passed by reference for ccall in any case ;)

I don't think wrapping in a `Ref` would work, but now that you mention that structs get passed by reference,  I suppose this is why it worked....

Reply:
>  structs get passed by reference for ccall 

is this documented?  I can't find it here https://docs.julialang.org/en/v1/manual/calling-c-and-fortran-code/  but I belive you